src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,570 @@
+from __future__ import annotations
+
+import datetime
+import os
+from dataclasses import dataclass
+from os import fspath
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import opera_utils as oput
+from osgeo import gdal
+from pyproj import CRS
+from scipy.interpolate import RegularGridInterpolator
+
+from dolphin import io, stitching
+from dolphin._dates import _format_date_pair, get_dates
+from dolphin._log import get_log
+from dolphin._types import Bbox, Filename, TropoModel, TropoType
+
+logger = get_log(__name__)
+
+###########
+
+__all__ = ["estimate_tropospheric_delay"]
+
+
+@dataclass
+class DelayParams:
+    """Parameters for estimating tropospheric delay corrections."""
+
+    x_coordinates: np.ndarray
+    """Array of X coordinates."""
+
+    y_coordinates: np.ndarray
+    """Array of Y coordinates."""
+
+    z_coordinates: np.ndarray
+    """Array of Z coordinates."""
+
+    bounds: Bbox
+    """ Bounding box of the data."""
```

Comment:
I think the last main comment I have for now is that we should change the name of this to `snwe`- i got confused when looked through the main PyAPS/Raider logic because the `Bbox` class to me always means (left, bottom, right, top) like it is in `shapely` and `rasterio`. I was actually thinking of changeing the `Bbox` type to a [`namedtuple` like how it is in `rasterio`](https://github.com/rasterio/rasterio/blob/5c441ed7480b33cff4c2071c9a0d446ace7c3cec/rasterio/coords.py#L5) so that its even more clear that the numbers mean "left, bottom, right, top".

Reply:
I agree, also namedtuple sounds good, I am not sure how much you need to change for that in other use cases